modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/resource/v1_0/ObjectFolderResourceImpl.java

```diff
@@ -243,16 +248,28 @@ private void _addObjectFolderResources(
 				objectDefinition.setObjectFolderExternalReferenceCode(
 					objectFolderExternalReferenceCode);
 
-				objectDefinition =
-					objectDefinitionResource.
-						putObjectDefinitionByExternalReferenceCode(
-							objectDefinition.getExternalReferenceCode(),
-							objectDefinition);
-
-				_objectFolderItemLocalService.updateObjectFolderItem(
-					objectDefinition.getId(), objectFolderId,
-					unlinkedObjectFolderItem.getPositionX(),
-					unlinkedObjectFolderItem.getPositionY());
+				try {
+					objectDefinition =
+						objectDefinitionResource.
+							putObjectDefinitionByExternalReferenceCode(
+								objectDefinition.getExternalReferenceCode(),
+								objectDefinition);
+
+					_objectFolderItemLocalService.updateObjectFolderItem(
+						objectDefinition.getId(), objectFolderId,
+						unlinkedObjectFolderItem.getPositionX(),
+						unlinkedObjectFolderItem.getPositionY());
+				}
+				catch (Exception exception) {
+					_log.error(exception);
+
+					failedObjectDefinitionNames.add(objectDefinition.getName());
+
+					objectDefinition =
+						objectDefinitionResource.
+							getObjectDefinitionByExternalReferenceCode(
+								objectDefinition.getExternalReferenceCode());
+				}
 
 				serviceBuilderObjectDefinitionIds.remove(
 					objectDefinition.getId());
```

Comment:
What happens if the `objectDefinition` is Null?
Shouldn't it be safer to verify before removing it from the list?

Reply:
objectDefinitionResource.getObjectDefinitionByExternalReferenceCode() is a Fetch and can return null.